### PR TITLE
PPI-979 new ecommerce methods for gatsby plugin

### PR DESCRIPTION
**Pull request by waszkowski**
- move the plugin source code to subdirectory `plugin`
- upgrade `@piwikpro/react-piwik-pro` to the newest version
- update examples for the eCommerce part
- update `README.md` for the eCommerce part

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -240,6 +240,29 @@ Collection of methods to handle eCommerce events through the Piwik PRO API.
...
+```
+
+- `ecommerceAddToCart(products: Product[])` - Tracks action of adding products to a cart.
+- `ecommerceRemoveFromCart(products: Product[])` - Tracks action of removing a products from a cart.
+- `ecommerceOrder(products: Product[], paymentInformation: PaymentInformation)` - Tracks conversion (including products and payment details).
+- `ecommerceCartUpdate(products: Product[], grandTotal: PaymentInformation['grandTotal'])` - Tracks current state of a cart.
+- `ecommerceProductDetailView(products: Product[])` - Tracks product or category view. Must be followed by a page view.
+
+Deprecated methods:
 - `eCommerce.addEcommerceItem(productSKU[, productName[, productCategory[, productPrice[, productQuantity]]]])` - Adds a product to a virtual shopping cart. If a product with the same SKU is in the cart, it will be removed first. Does not send any data to the Collecting & Processing Pipeline.
```
Comment (lysy-vlc) [inline README.md:266]: Consider removing deprecated methods from documetation

File: example/package.json [hunk 2]
```diff
@@ -13,10 +13,9 @@
     "clean": "gatsby clean"
   },
   "dependencies": {
-    "@piwikpro/gatsby-plugin-piwik-pro": "file:../",
+    "@piwikpro/gatsby-plugin-piwik-pro": "file:../plugin",
```
Comment (lysy-vlc) [inline example/package.json:16]: how to resolve this from package downloaded from npm?

File: example/src/pages/ContentTracking.tsx [hunk 3]
```diff
@@ -56,7 +56,7 @@ const ContentTrackingPage = () => {
         <h1>{pageData.title}</h1>
         <p>{pageData.description}</p>
         <h1>Import</h1>
-        <code>{`import { ContentTracking } from '@piwikpro/gatsby-piwik-pro';`}</code>
+        <code>{`import { ContentTracking } from '@piwikpro/gatsby-plugin-piwik-pro';`}</code>
```
Comment (lysy-vlc) [inline example/src/pages/ContentTracking.tsx:59]: Please check out readme or npm page. There is a mismatch in naming

**Review by lysy-vlc — CHANGES_REQUESTED** [review]

File: example/package.json (hunk 2 above)
Comment (waszkowski) [reply example/package.json:16]: I'm not sure what you mean. If you want to use this package from npm, you should use `@piwikpro/gatsby-plugin-piwik-pro": "1.0.0"`.

File: example/src/pages/ContentTracking.tsx (hunk 3 above)
Comment (waszkowski) [reply example/src/pages/ContentTracking.tsx:59]: What's wrong?

File: README.md (hunk 1 above)
Comment (waszkowski) [reply README.md:266]: I think we should leave these descriptions until the methods are removed from the package.

Comment (danieltwork) [reply README.md:266]: I am of the same opinion as @waszkowski

**Review by danieltwork — APPROVED** [review]

**Review by lysy-vlc — APPROVED** [review]

**Pull request merged**